core/src/client/lwm2m_static.c

```diff
@@ -318,6 +318,93 @@ int AwaStaticClient_Process(AwaStaticClient * client)
     return result;
 }
 
+static AwaLwm2mResult AwaStaticClientDefaultHandler(AwaStaticClient * client, AwaOperation operation, AwaObjectID objectID, 
+                                                    AwaObjectInstanceID objectInstanceID, AwaResourceID resourceID, AwaResourceInstanceID resourceInstanceID, 
+                                                    void ** dataPointer, uint16_t * dataSize, bool * changed)
+{
+    AwaLwm2mResult result;
+
+    ObjectDefinition * objectDefinition = Definition_LookupObjectDefinition(Lwm2mCore_GetDefinitions(client->Context), objectID);
+    if (objectDefinition != NULL)
+    {
+        uint8_t * offset;
+        ResourceDefinition * resourceDefinition;
+
+        // check instance range
+        if ((objectInstanceID >= 0) && (objectInstanceID < objectDefinition->MaximumInstances))
+        {
+            switch (operation)
+            {
+            case AwaOperation_CreateObjectInstance:
+                result = AwaLwm2mResult_SuccessCreated;
+                break;
+
+            case AwaOperation_DeleteObjectInstance:
+                result = AwaLwm2mResult_SuccessDeleted;
+                break;
+
+            case AwaOperation_CreateResource:
+                result = AwaLwm2mResult_SuccessCreated;
+                break;
+
+            case AwaOperation_DeleteResource:
+                break;
+
+            case AwaOperation_Write:
+            
+                resourceDefinition = Definition_LookupResourceDefinitionFromObjectDefinition(objectDefinition, resourceID);
+                if (resourceDefinition != NULL)
+                {
+                // TODO: check dataSize vs definition->DataElementSize
+                // TODO: what do we do about storing the length.. for opaque etc...
+                    offset = resourceDefinition->DataPointer + (resourceDefinition->DataStepSize * objectInstanceID);
+                    memcpy(offset, *dataPointer, *dataSize);
+                    result = AwaLwm2mResult_SuccessChanged;
+                }
+                else
+                {
+                    result = AwaLwm2mResult_BadRequest;
+                }
+                break;
+
+            case AwaOperation_Read:
+
+                resourceDefinition = Definition_LookupResourceDefinitionFromObjectDefinition(objectDefinition, resourceID);
+                if (resourceDefinition != NULL)
+                {
+                    offset = resourceDefinition->DataPointer + (resourceDefinition->DataStepSize * objectInstanceID);
```

Comment:
There are 3 parameters in AwaStaticClient_RegisterResourceWithPointer() the dataPointer, dataElementSize and dataStepSize. The code above assumes that the step size will be set to the element size for sequential data (i.e array types) and will be set to the sizeof the structure when using structures. 

for example: 

`typedef struct
{
    char Manufacturer[64];
    float Temperature;

} HeaterObject;

static HeaterObject heater[HEATER_INSTANCES];

AwaStaticClient_RegisterObject(awaClient, "Heater", 1000, 0, HEATER_INSTANCES);
AwaStaticClient_RegisterResourceWithPointer(awaClient, "Manufacturer", 1000, 101, AwaResourceType_String, 0, 1, AwaAccess_Read,
                                                &heater[0].Manufacturer, sizeof(heater[0].Manufacturer), sizeof(heater[0]));
AwaStaticClient_RegisterResourceWithPointer(awaClient, "Temperature",  1000, 104, AwaResourceType_Float, 0, 1, AwaAccess_Read,
                                                &heater[0].Temperature, sizeof(heater[0].Temperature), sizeof(heater[0]));

`